isototest/src/action.rs

```diff
@@ -77,73 +83,99 @@ pub async fn write_to_console(client: &VncClient, text: &str) -> Result<(), VncE
-        '4' => Some(VirtualKeyCode::Key4),
-        '5' => Some(VirtualKeyCode::Key5),
-        '6' => Some(VirtualKeyCode::Key6),
-        '7' => Some(VirtualKeyCode::Key7),
-        '8' => Some(VirtualKeyCode::Key8),
-        '9' => Some(VirtualKeyCode::Key9),
-        '0' => Some(VirtualKeyCode::Key0),
-        'a' | 'A' => Some(VirtualKeyCode::A),
-        'b' | 'B' => Some(VirtualKeyCode::B),
-        'c' | 'C' => Some(VirtualKeyCode::C),
-        'd' | 'D' => Some(VirtualKeyCode::D),
-        'e' | 'E' => Some(VirtualKeyCode::E),
-        'f' | 'F' => Some(VirtualKeyCode::F),
-        'g' | 'G' => Some(VirtualKeyCode::G),
-        'h' | 'H' => Some(VirtualKeyCode::H),
-        'i' | 'I' => Some(VirtualKeyCode::I),
-        'j' | 'J' => Some(VirtualKeyCode::J),
-        'k' | 'K' => Some(VirtualKeyCode::K),
-        'l' | 'L' => Some(VirtualKeyCode::L),
-        'm' | 'M' => Some(VirtualKeyCode::M),
-        'n' | 'N' => Some(VirtualKeyCode::N),
-        'o' | 'O' => Some(VirtualKeyCode::O),
-        'p' | 'P' => Some(VirtualKeyCode::P),
-        'q' | 'Q' => Some(VirtualKeyCode::Q),
-        'r' | 'R' => Some(VirtualKeyCode::R),
-        's' | 'S' => Some(VirtualKeyCode::S),
-        't' | 'T' => Some(VirtualKeyCode::T),
-        'u' | 'U' => Some(VirtualKeyCode::U),
-        'v' | 'V' => Some(VirtualKeyCode::V),
-        'w' | 'W' => Some(VirtualKeyCode::W),
-        'x' | 'X' => Some(VirtualKeyCode::X),
-        'y' | 'Y' => Some(VirtualKeyCode::Y),
-        'z' | 'Z' => Some(VirtualKeyCode::Z),
-        ' ' => Some(VirtualKeyCode::Space),
-        '!' => Some(VirtualKeyCode::Key1),
-        '@' => Some(VirtualKeyCode::Key2),
-        '#' => Some(VirtualKeyCode::Key3),
-        '$' => Some(VirtualKeyCode::Key4),
-        '%' => Some(VirtualKeyCode::Key5),
-        '^' => Some(VirtualKeyCode::Caret),
-        '&' => Some(VirtualKeyCode::Key7),
-        '*' => Some(VirtualKeyCode::Key8),
-        '(' => Some(VirtualKeyCode::Key9),
-        ')' => Some(VirtualKeyCode::Key0),
-        '-' => Some(VirtualKeyCode::Minus),
-        '_' => Some(VirtualKeyCode::Underline),
-        '=' => Some(VirtualKeyCode::Equals),
-        '+' => Some(VirtualKeyCode::Plus),
-        '[' => Some(VirtualKeyCode::LBracket),
-        ']' => Some(VirtualKeyCode::RBracket),
-        '{' => Some(VirtualKeyCode::LBracket),
-        '}' => Some(VirtualKeyCode::RBracket),
-        '\\' => Some(VirtualKeyCode::Backslash),
-        '|' => Some(VirtualKeyCode::Backslash),
-        ';' => Some(VirtualKeyCode::Semicolon),
-        ':' => Some(VirtualKeyCode::Colon),
-        '\'' => Some(VirtualKeyCode::Apostrophe),
-        '"' => Some(VirtualKeyCode::Apostrophe),
-        ',' => Some(VirtualKeyCode::Comma),
-        '.' => Some(VirtualKeyCode::Period),
-        '/' => Some(VirtualKeyCode::Slash),
-        '<' => Some(VirtualKeyCode::Comma),
-        '>' => Some(VirtualKeyCode::Period),
-        '?' => Some(VirtualKeyCode::Slash),
+        '1' => Some(KeyCode::Key1),
```

Comment:
who provides `KeyCode`? VNC? - you can also check whats missing from: https://github.com/os-autoinst/os-autoinst/blob/master/consoles/VNC.pm#L495

Reply:
no, `KeyCode` is an `enum` I wrote which returns the character's corresponing ASCII code

Reply:
Maybe the mask approach is wrong? 
https://www.x.org/releases/current/doc/xproto/x11protocol.html#keysym_encoding